### PR TITLE
fix: refactoring parser to address long response bugs and debug log cleanups

### PR DESCRIPTION
**Pull request by sfc-gh-alherrera**
…g logs cleanup

### CODE REVIEW

File: agent_gateway/gateway/gateway.py [hunk 1]
```diff
@@ -50,20 +51,20 @@ def __init__(self, session, llm) -> None:
...
-        gateway_logger.log(logging.DEBUG, "Cortex Request Data\n", data, block=True)
-
-        response_text = await post_cortex_request(url=url, headers=headers, data=data)
-        gateway_logger.log(
-            logging.DEBUG,
-            "Cortex Request Response\n",
-            response_text,
-            block=True,
-        )
+        # gateway_logger.log(logging.DEBUG, "Cortex LLM Prompt\n", data, block=True)
```
Comment (sfc-gh-twhite) [inline agent_gateway/gateway/gateway.py:54]:
```suggestion
```

Let's go ahead and pull this comment out.

File: agent_gateway/gateway/gateway.py [hunk 2]
```diff
@@ -50,20 +51,20 @@ def __init__(self, session, llm) -> None:
...
+            )
+        except Exception as e:
+            raise AgentGatewayError(
+                message=f"Failed Cortex LLM Request. See details:{str(e)}"
+            ) from e
 
         if "choices" not in response_text:
             raise AgentGatewayError(
-                message=f"Failed Cortex LLM Request. Missing choices in response. See details:{response_text}"
+                message=f"Invalid Cortex LLM Response. See details:{response_text}"
```
Comment (sfc-gh-twhite) [inline agent_gateway/gateway/gateway.py:66]: What are the possibilities here?

Would it only hit this block in the case of Analyst not returning either the text response with suggestions or an SQL expression?

**Review by sfc-gh-twhite — APPROVED** [review]
LGTM! Small comment regarding clarity (maybe we can add some comments about why we're inside that "if choices not in" section) - but not a blocker.

File: agent_gateway/gateway/gateway.py (hunk 2 above)
Comment (sfc-gh-alherrera) [reply agent_gateway/gateway/gateway.py:66]: this is for the complete calls, not analyst. there can be cases where complete is not configured properly (invalid model provided by end user..etc) or the response comes back empty

**Pull request merged**